Add voice cheat sheet and fix docs typos (#590)
Kraigie/nostrum: guides/functionality/voice.md

When `Nostrum.Voice.play/4` is called with `:ytdl` for the `type` parameter, `youtube-dl` will be
run with options `-f bestaudio -q -o -`, which will attempt to download the audio at the given url and pipe it to `ffmpeg`.

> #### Forks {: .warning}

> > More recently `youtube-dl` hasn't been maintained regularly enough to keep up with YouTube's changes.
> > The use of the `yt-dlp` fork is recommended in its place.

Reviewer: ```suggestion
> Recent versions of `youtube-dl` have not been maintained regularly enough to adopt recent YouTube changes.
> The use of the `yt-dlp` fork is recommended in its place.
```

Author: This implies that there are recent version of `youtube-dl` which there are not (latest is from 2021) and uses "recent" twice in the same sentence 😁 
I've reworded it.

>

> > `config :nostrum, :youtubedl, "yt-dlp"`

Reviewer: Change this into an actual \`\`\`elixir codeblock.
